domain/model/service/modelservice.go

```diff
@@ -20,16 +21,23 @@ type ModelState interface {
 	Delete(ctx context.Context, uuid coremodel.UUID) error
 }
 
+// ModelsState represents the state required for reading all model information.
+type ModelsState interface {
+	Get(context.Context, coremodel.UUID) (coremodel.Model, error)
+}
+
 // ModelService defines a service for interacting with the underlying model
 // state, as opposed to the controller state.
 type ModelService struct {
-	st ModelState
+	modelsST ModelsState
+	st       ModelState
 }
 
 // NewModelService returns a new Service for interacting with a models state.
-func NewModelService(st ModelState) *ModelService {
+func NewModelService(modelsST ModelsState, st ModelState) *ModelService {
```

Comment:
This then becomes `modelGetter ModelGetter` which is more intuitive.

Reply:
Done & Done